From 2ed88827aceb9d20a8c51ffdadd9fdd0ee9eab47 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Tue, 9 Apr 2019 10:22:13 +0800
Subject: [PATCH 14/30] media: imx8: mxc-mipi-csi2_yav: fix compile error

In upstream commit b6ee3f0dcf43 ("media: v4l: async: Move async subdev
notifier operations to a separate structure"), notifier.bound has been
moved into "struct v4l2_async_notifier_operations".

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/media/platform/imx8/mxc-media-dev.c       | 7 +++++--
 drivers/media/platform/imx8/mxc-mipi-csi2_yav.c   | 6 +++++-
 drivers/media/platform/mxc/capture/mxc_mipi_csi.c | 6 +++++-
 3 files changed, 15 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/imx8/mxc-media-dev.c b/drivers/media/platform/imx8/mxc-media-dev.c
index 52d3804b927b..344748c2b52c 100644
--- a/drivers/media/platform/imx8/mxc-media-dev.c
+++ b/drivers/media/platform/imx8/mxc-media-dev.c
@@ -627,6 +627,10 @@ static const struct media_device_ops mxc_md_ops = {
 	.link_notify = mxc_md_link_notify,
 };
 
+static const struct v4l2_async_notifier_operations mxc_md_async_ops = {
+	.bound = subdev_notifier_bound,
+	.complete = subdev_notifier_complete,
+};
 
 static int mxc_md_probe(struct platform_device *pdev)
 {
@@ -678,8 +682,7 @@ static int mxc_md_probe(struct platform_device *pdev)
 	if (mxc_md->num_sensors > 0) {
 		mxc_md->subdev_notifier.subdevs = mxc_md->async_subdevs;
 		mxc_md->subdev_notifier.num_subdevs = mxc_md->num_sensors;
-		mxc_md->subdev_notifier.bound = subdev_notifier_bound;
-		mxc_md->subdev_notifier.complete = subdev_notifier_complete;
+		mxc_md->subdev_notifier.ops = &mxc_md_async_ops;
 		mxc_md->num_sensors = 0;
 
 		ret = v4l2_async_notifier_register(&mxc_md->v4l2_dev,
diff --git a/drivers/media/platform/imx8/mxc-mipi-csi2_yav.c b/drivers/media/platform/imx8/mxc-mipi-csi2_yav.c
index abeb38279962..df5710aa186b 100644
--- a/drivers/media/platform/imx8/mxc-mipi-csi2_yav.c
+++ b/drivers/media/platform/imx8/mxc-mipi-csi2_yav.c
@@ -513,6 +513,10 @@ static int subdev_notifier_bound(struct v4l2_async_notifier *notifier,
 	return 0;
 }
 
+static const struct v4l2_async_notifier_operations mxc_mipi_async_ops = {
+		.bound = subdev_notifier_bound,
+};
+
 static int mipi_csis_subdev_host(struct mxc_mipi_csi2_dev *csi2dev)
 {
 	struct device *dev = &csi2dev->pdev->dev;
@@ -551,7 +555,7 @@ static int mipi_csis_subdev_host(struct mxc_mipi_csi2_dev *csi2dev)
 
 	csi2dev->subdev_notifier.subdevs = csi2dev->async_subdevs;
 	csi2dev->subdev_notifier.num_subdevs = 1;
-	csi2dev->subdev_notifier.bound = subdev_notifier_bound;
+	csi2dev->subdev_notifier.ops = &mxc_mipi_async_ops;
 
 	ret = v4l2_async_notifier_register(&csi2dev->v4l2_dev,
 					   &csi2dev->subdev_notifier);
diff --git a/drivers/media/platform/mxc/capture/mxc_mipi_csi.c b/drivers/media/platform/mxc/capture/mxc_mipi_csi.c
index a7964b7b8345..30860ac6610b 100644
--- a/drivers/media/platform/mxc/capture/mxc_mipi_csi.c
+++ b/drivers/media/platform/mxc/capture/mxc_mipi_csi.c
@@ -993,6 +993,10 @@ static int mipi_csis_parse_dt(struct platform_device *pdev,
 static int mipi_csis_pm_resume(struct device *dev, bool runtime);
 static const struct of_device_id mipi_csis_of_match[];
 
+static const struct v4l2_async_notifier_operations mipi_csis_async_ops = {
+	.bound = subdev_notifier_bound,
+};
+
 /* register parent dev */
 static int mipi_csis_subdev_host(struct csi_state *state)
 {
@@ -1028,7 +1032,7 @@ static int mipi_csis_subdev_host(struct csi_state *state)
 
 	state->subdev_notifier.subdevs = state->async_subdevs;
 	state->subdev_notifier.num_subdevs = 1;
-	state->subdev_notifier.bound = subdev_notifier_bound;
+	state->subdev_notifier.ops = &mipi_csis_async_ops;
 
 	ret = v4l2_async_notifier_register(&state->v4l2_dev,
 					&state->subdev_notifier);
-- 
2.17.1

